Sources/Workspace/UserToolchain.swift

```diff
@@ -302,19 +308,20 @@ public final class UserToolchain: Toolchain {
 
     // MARK: - initializer
 
-    public init(destination: Destination, environment: EnvironmentVariables = .process()) throws {
+    public init(destination: Destination, environment: EnvironmentVariables = .process(), explicitSearchPaths: [AbsolutePath]? = nil) throws {
         self.destination = destination
 
         // Get the search paths from PATH.
-        self.envSearchPaths = getEnvSearchPaths(
+        self.envSearchPaths = explicitSearchPaths ?? getEnvSearchPaths(
             pathString: ProcessEnv.path,
             currentWorkingDirectory: localFileSystem.currentWorkingDirectory
         )
+        self.searchPathsOnly = explicitSearchPaths != nil
```

Comment:
I wonder if this should be its own argument instead of derived from the customSearchPaths

Reply:
```
public init(destination: Destination, environment: EnvironmentVariables = .process(), customSearchPaths: [AbsolutePath]? = .none, useXcrun: Bool = true) throws
```

Reply:
What would that mean in practice? If not found in search paths, fall back to `xcrun`? Or if `xcrun` errors, we ignore it and look at search paths? I think either could make sense, but offering both seems like a little more complexity that is maybe needed.

Reply:
maybe the solution is an enumeration? ie you can say: use xcrun or searchnin default path, or search in custom path? and the xcrun enum case can be available only on macOS

Reply:
So I think we just have a binary choice here, right? Either default or custom paths, it's just that default has a different meaning on macOS vs. other platforms. On macOS, default means `xcrun` and on other platforms it means default search paths.